### PR TITLE
Test analysis updates

### PR DESCRIPTION
**Pull request by stbenjam**
Shows per-variant results on the test anaylsis pages

### CODE REVIEW

**Review by dgoodwin — COMMENTED** [review]
Tested locally and this looks amazing, massive help.

I'm seeing some weirdness locally with a copy of the prod db, very very slow requests and queries, but once you run one it seems to get faster for a bit, then I leave it be and it'll be slower the next request I make again. 

```
2022/05/18 15:57:14 /home/dgoodwin/go/src/github.com/openshift/sippy/pkg/api/test_analysis.go:156 SLOW SQL >= 200ms                                                                                                                    
[126853.222ms] [rows:56] SELECT test_id,                                                                                                                                                                                               
               test_name,                                                                                                                                                                                                              
           date,                                                                                                                                                                                                                       
               release,                                                                                                                                                                                                                
               variant,                                                                                                                                                                                                                
               runs,                                                                                                                                                                                                                   
               passes,                                                                                                                                                                                                                 
               flakes,                                                                                                                                                                                                                 
           failures FROM "prow_test_analysis_by_variant_14d_matview" WHERE release = '4.11' AND test_name = 'BackendDisruption.service-load-balancer-with-pdb-reused-connections zero-disruption should not be worse' AND variant NOT I
N ('never-stable','techpreview') GROUP BY test_id, test_name, date, release, variant, runs, passes, flakes, failures                                                                                                                   
                                                                                                                                       
example: INFO[2022-05-18T15:57:14.602-03:00] responded to request                          elapsed=2m6.889981409s method=GET uri="/api/tests/analysis?release=4.11&test=BackendDisruption.service-load-balancer-with-pdb-reused-connections%20ze
ro-disruption%20should%20not%20be%20worse&filter=%7B%22items%22%3A%5B%7B%22columnField%22%3A%22name%22%2C%22operatorValue%22%3A%22equals%22%2C%22value%22%3A%22BackendDisruption.service-load-balancer-with-pdb-reused-connections%20ze
ro-disruption%20should%20not%20be%20worse%22%7D%2C%7B%22columnField%22%3A%22variants%22%2C%22operatorValue%22%3A%22contains%22%2C%22value%22%3A%22never-stable%22%2C%22not%22%3Atrue%7D%2C%7B%22columnField%22%3A%22variants%22%2C%22op
eratorValue%22%3A%22contains%22%2C%22value%22%3A%22techpreview%22%2C%22not%22%3Atrue%7D%5D%7D"
```

Once that responded I ran again and it was quite fast, 790ms for the whole request. I had no matview refreshes running at the time as I've disabled them locally.

Very unlikely from this PR and I think I've been seeing it in prod.

File: pkg/api/test_analysis.go [hunk 1]
```diff
@@ -103,56 +104,88 @@ func PrintTestAnalysisJSON(w http.ResponseWriter, req *http.Request, curr, prev
 	RespondWithJSON(http.StatusOK, w, results)
 }
 
-func PrintTestAnalysisJSONFromDB(dbc *db.DB, w http.ResponseWriter, release, testName string) error {
+func PrintTestAnalysisJSONFromDB(dbc *db.DB, w http.ResponseWriter, req *http.Request, release, testName string) error {
```
[the remark this answers is not in the document]
Comment (stbenjam) [reply pkg/api/test_analysis.go:107]: I think that makes sense, I hadn't really considered these things might be accessed not via the JSON API.  Should all of the param parsing and response handing be done in the server.go side?

**Comment by stbenjam** [comment]
> I'm seeing some weirdness locally with a copy of the prod db, very very slow requests and queries, but once you run one it seems to get faster for a bit, then I leave it be and it'll be slower the next request I make again.

I noticed this. It does happen for sure when materialized views are being updated.  If it's happening outside of that, maybe something with go garbage collection, or something going on in postgres.  Maybe we can add some instrumentation to figure out what's going on? I can create a jira card for that.

File: sippy-ng/src/tests/TestAnalysis.js [hunk 2]
```diff
@@ -383,49 +398,120 @@ export function TestAnalysis(props) {
...
+              <Typography variant="h5">
+                Pass Rate By NURP+ Combination
+                <Tooltip
+                  title={
+                    <p>
+                      NURP+ is the combination of a job&apos;s <b>N</b>etwork
+                      (e.g. sdn, ovn), <b>U</b>pgrade, <b>R</b>elease (minor or
+                      micro),
+                      <b>P</b>latform (aws, azure, etc) and extras (realtime,
+                      serial, etc)
```
[the remark this answers is not in the document]
Comment (dgoodwin) [reply sippy-ng/src/tests/TestAnalysis.js:501]: Ah my mistake, yeah "From Release" would probably help, Release is so prominent in sippy that threw me for a loop.

**Comment by dgoodwin** [comment]
Yeah that's a good idea (the jira), maybe we can pin it down to specific APIs or even specific queries somehow, if we could come up with an identifier for a label.

File: pkg/api/test_analysis.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgoodwin) [reply pkg/api/test_analysis.go:107]: Yeah that would give that layer a good purpose and set us up for better reusability.

**Comment by dgoodwin** [comment]
Let me know when this is ready again, I'd like to see the 14 days clarified in the tooltip, the server.go layering stuff can wait if need  be.

**Pull request merged**